components/dropinblog/sources/common/polling.mjs

```diff
@@ -0,0 +1,55 @@
+import {
+  ConfigurationError,
+  DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+} from "@pipedream/platform";
+import app from "../../dropinblog.app.mjs";
+
+export default {
+  props: {
+    app,
+    timer: {
+      type: "$.interface.timer",
+      label: "Polling Schedule",
+      description: "How often to poll the API",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+  },
+  methods: {
+    generateMeta() {
+      throw new ConfigurationError("generateMeta is not implemented");
+    },
+    getResourceName() {
+      throw new ConfigurationError("getResourceName is not implemented");
+    },
+    getResourcesFn() {
+      throw new ConfigurationError("getResourcesFn is not implemented");
+    },
+    getResourcesFnArgs() {
+      throw new ConfigurationError("getResourcesFnArgs is not implemented");
+    },
+    processResource(resource) {
+      const meta = this.generateMeta(resource);
+      this.$emit(resource, meta);
+    },
+  },
+  async run() {
+    const {
+      app,
+      getResourcesFn,
+      getResourcesFnArgs,
+      getResourceName,
+      processResource,
+    } = this;
+
+    const resources = await app.paginate({
+      resourcesFn: getResourcesFn(),
+      resourcesFnArgs: getResourcesFnArgs(),
+      resourceName: getResourceName(),
+    });
+
+    Array.from(resources)
+      .forEach(processResource);
```

Comment:
This will emit a max of 100 events (historical events) on the first run. We should list no more than 25 historical events, and include a `test-event.mjs` if possible.

Using `dedupe:unique` will dedupe up to 1000 ids, but we should avoid using it as the only deduping strategy.  I would check each post's `publishedAtIso8601` date to make sure it's greater than the last one we've emitted.

Reply:
If the events are ordered (I'm not sure if they are in this case), you could emit the 25 most recently published posts and then set the `lastPublishedDate`. If they're not ordered, you could still loop through them all to find the `lastPublishedDate`, but only emit 25 of them. Either way, the `run` method won't emit any more posts because it will be checking the `lastPublishedDate`.